bindgen/callbacks.rs

```diff
@@ -120,10 +121,13 @@ pub trait ParseCallbacks: fmt::Debug {
 
 /// Relevant information about a type to which new derive attributes will be added using
 /// [`ParseCallbacks::add_derives`].
+#[derive(Debug)]
 #[non_exhaustive]
 pub struct DeriveInfo<'a> {
     /// The name of the type.
     pub name: &'a str,
+    /// The kind of the type
+    pub kind: Option<CompKind>,
```

Comment:
I think this shoudn't be wrapped in an `Option` but the use should be able to ignore the `kind` if they want to from the callback implementation directly:

```rust
    fn add_derives(&self, info: &bindgen::callbacks::DeriveInfo<'_>) -> Vec<String> {
        let mut derives = vec![];

        match &info.kind {
            CompKind::Struct => /* add derives exclusive for `struct`s */,
            CompKind::Union => /* add derives exclusive for `union`s */,
        }

        /* add derives for any kind of type */
        
        derives
    }
```

Reply:
Well I wasn't regexing on CompKind values, and there's no `Any` member of that enum, so I figured that if you wanted them all, we could use None, and if you wanted none then you shouldn't be using the option 😄 

This was to allow for the other branch where I don't filter by CompKind because it is not available.  I did not want to stop all over people wanting to derive things on other rust types that support it.

Reply:
I find the `None` confusing because if someone does something like:

```rust
    fn add_derives(&self, info: &bindgen::callbacks::DeriveInfo<'_>) -> Vec<String> {
        if info.kind.is_none() {
            vec!["serde::Serialize"]
        } else {
            vec![]
        }
    }
```

Then `Serialize` would only be derived for rust `enum`s instead of being for "every" kind of type definition.